src/deferred.js

```diff
@@ -13,6 +13,23 @@ function Thrower( ex ) {
 	throw ex;
 }
 
+function adoptValue( value, resolve, reject ) {
+	var method;
+
+	// Check for promise aspect first to privilege synchronous behavior
+	if ( value && jQuery.isFunction( ( method = value.promise ) ) ) {
```

Comment:
So assignment triggers getter, `isFunction` also does it if called directly on object and does not trigger it if called indirectly. 

So in above example we would have same amount of triggered accessors i.e. i don't see violations here